api/models/Post.js

```diff
@@ -90,6 +90,18 @@ module.exports = bookshelf.Model.extend({
     return this.belongsTo(LinkPreview)
   },
 
+  childTags: function (opts) {
+    // this is part of the 'taggable' interface, shared with Comment
+    return this.load(['comments', 'comments.tags'], opts)
```

Comment:
and then you could do `uniqBy('id', flatten(p.relations.comments.map(c => c.relations.tags.models)))`